core/src/test/scala/org/broadinstitute/dsde/rawls/spendreporting/SpendReportingServiceSpec.scala

```diff
@@ -67,16 +99,33 @@ class SpendReportingServiceSpec extends AnyFlatSpecLike with TestDriverComponent
     new SpendReportingService(userInfo, dataSource, mockServiceFactory.getServiceFromJson("json", defaultServiceProject), samDAO, spendReportingServiceConfig)
   }
 
-  "SpendReportingService" should "unmarshal results from Google" in withDefaultTestDatabase { dataSource: SlickDataSource =>
+  "SpendReportingService" should "break down results from Google by day" in withDefaultTestDatabase { dataSource: SlickDataSource =>
     val service = createSpendReportingService(dataSource)
 
-    val reportingResults = Await.result(service.getSpendForBillingProject(testData.billingProject.projectName, DateTime.now().minusDays(1), DateTime.now()), Duration.Inf)
+    val reportingResults = Await.result(service.getSpendForBillingProject(testData.billingProject.projectName, DateTime.now().minusDays(1), DateTime.now(), SpendReportingAggregationKeys.Daily), Duration.Inf)
     reportingResults.spendSummary.cost shouldBe "0.10" // sum of costs in defaultTable
+    reportingResults.spendDetails.headOption.getOrElse(fail("daily results not parsed correctly"))
+      .aggregationKey shouldBe SpendReportingAggregationKeys.Daily
+  }
+
+  it should "break down results from Google by workspace" in withDefaultTestDatabase { dataSource: SlickDataSource =>
+    val service = createSpendReportingService(dataSource, tableResult = workspaceTableResult)
+
+    runAndWait(dataSource.dataAccess.workspaceQuery.createOrUpdate(workspace1))
+    runAndWait(dataSource.dataAccess.workspaceQuery.createOrUpdate(workspace2))
+
+    val reportingResults = Await.result(service.getSpendForBillingProject(testData.billingProject.projectName, DateTime.now().minusDays(1), DateTime.now(), SpendReportingAggregationKeys.Workspace), Duration.Inf)
+    reportingResults.spendSummary.cost shouldBe "0.10" // sum of costs in defaultTable
+    val workspaceAggregation = reportingResults.spendDetails.headOption.getOrElse(fail("workspace results not parsed correctly"))
+
+    workspaceAggregation.aggregationKey shouldBe SpendReportingAggregationKeys.Workspace
+    workspaceAggregation.spendData.headOption.getOrElse(fail("workspace results not parsed correctly")).googleProjectId shouldBe defined
```

Comment:
It seems like it might be worth testing that the cost information for each workspace is correct (and have non-zero costs for both so that the sum is actually the sum of both things).

I have the same comment for the date aggregation.